### PR TITLE
Expose cache statistics endpoints on authenticated portals

### PR DESCRIPTION
**Pull request by pvannierop**
# Problem

On SAML-authenticated portals, the cache statistics endpoints are not accessible programatically.

# Solution

This PR will remove the cache statistics endpoints from Spring Security protection.

# Remark

The cache statistics endpoints are inactive by default and can be activated with the `cache.statistics_enpoint_enable` property. Because users make the concious decision to enable the endpoint, I see no security risk in exposing it.

### CODE REVIEW

**Comment by jjgao** [comment]
@pvannierop @n1zea144 @sheridancbio is this still relevant?

**Comment by pvannierop** [comment]
@jjgao Yes, it is.

**Comment by jjgao** [comment]
@sheridancbio @n1zea144 would you please help review this?

**Review by inodb — APPROVED** [review]
LGTM

File: security/security-spring/src/main/resources/applicationContext-security.xml [hunk 1]
```diff
@@ -88,6 +88,9 @@
     <http pattern="/api/swagger-ui.html" security="none"/>
     <http pattern="/api/health" security="none"/>
     <http pattern="/api/cache/**" security="none"/>
+    <http pattern="/api/cacheStatistics" security="none"/>
+    <http pattern="/api/**/keysInCache" security="none"/>
+    <http pattern="/api/**/keyCountsPerClass" security="none"/>
```
Comment (inodb) [inline security/security-spring/src/main/resources/applicationContext-security.xml:93]: @pvannierop @dianab0 i resolved the conflicts with https://github.com/cBioPortal/cbioportal/pull/9046. Does this look good? Do we still need these other ones?

Comment (pvannierop) [reply security/security-spring/src/main/resources/applicationContext-security.xml:93]: @inodb This looks good!

**Pull request merged**